gap/attr.gi

```diff
@@ -954,6 +954,34 @@ function(digraph)
   return girth;
 end);
 
+InstallMethod(DigraphOddGirth, "for a digraph",
+[IsDigraph],
+function(digraph)
+  local A, B, k, N, girth, powerfound;
+  girth := DigraphGirth(digraph);
+  if girth = infinity or IsOddInt(girth) then
+    return girth;
+  fi;
+  N          := DigraphNrVertices(digraph);
+  A          := AdjacencyMatrix(digraph);
+  B          := Immutable(A ^ 2);
+  k          := 1;
+  powerfound := false;
+  while k <= N + 2 do
+    A := A * B;
+    k := k + 2;
+    if Trace(A) <> 0 then
+      # It suffices to find the trace as entries of A are positive.
+      powerfound := true;
+      break;
```

Comment:
The variable powerfound and this break aren’t necessary, you can just return k here, please amend!

Reply:
Also didn’t we discuss doing the powering thing by strongly connected component?